samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentGradingFacadeQueries.java

```diff
@@ -3075,11 +3076,13 @@ public int autoSubmitAssessments() {
                         retractDate = assessmentExtended.getRetractDate() != null ? assessmentExtended.getRetractDate() : retractDate;
                     }
 
-                    //If the due date or retract date hasn't passed yet, go on to the next one, don't consider it yet
-                    if ((AssessmentAccessControlIfc.ACCEPT_LATE_SUBMISSION.toString().equals(lateHandling) && retractDate!=null && (currentTime.before(retractDate) || adata.getAttemptDate().after(retractDate)))
-                            || (dueDate != null && currentTime.before(dueDate))) {
+                    // If the due date or retract date hasn't passed yet, go on to the next one, don't consider it yet
+                    if (acceptLate && retractDate != null && (currentTime.before(retractDate) || adata.getAttemptDate().after(retractDate))) {
```

Comment:
@bgarciaentornos or @bjones86 

I don't understand this one: adata.getAttemptDate().after(retractDate)

Why would an attempt happen after retractDate?

Reply:
Could it be this? https://jira.sakaiproject.org/browse/SAK-36804 

"If auto-submit runs after excepted due date while excepted student is in quiz, quiz is not auto-submitted"

Reply:
Seems like you added it in 

SAM-3313 autosubmit job needs to process excepted late allowed submissions (#5024)

Reply:
Yeah, but I think I extracted it from the query and added the Java condition instead because otherwise more cases than due were being filtered. I can't remember it very well tbh, but I'd say it can happen, maybe settings being modified afterwards?

Reply:
I'm not sure, it seems a little strange. If the settings were modified to allow further submissions, I would think this would involve extending the retract date also, so I'm not sure how you could get a submission attempt date after the retract date. Maybe exceptions?

Reply:
Yeah I keep thinking it was just a crude way to sense an exception? I don't know if we should remove it. What do you guys think? I can't say I've ever come across a submission that was not submitted because of this clause. But it doesn't look right either.

Reply:
Yeah, I'm a little torn. On one hand, we haven't seen issues that this condition causes, but yet it doesn't seem like it should be there logically.

Reply:
Okay, let's leave it unless Bernardo thinks it should be removed.

Can I get an approval on this PR? It fixes an important bug.

Reply:
Yes, I'd leave it. And sorry for not approving it before, thought some other reviewer had already done it!